File: ChipmeoPOS/src/lib/types/role.ts
export interface Role {
	id: number;
	name: string;
	description?: string;
	defaultRoute?: string;
	isActive: boolean;
	createdAt?: string;
	updatedAt?: string;
}

export interface RoleCreateDto {
	name: string;
	description?: string;
	defaultRoute?: string;
	isActive: boolean;
}

export interface RoleUpdateDto {
	name: string;
	description?: string;
	defaultRoute?: string;
	isActive: boolean;
}

File: ChipmeoApis/ChipmeoApis.Core/Entities/Role.cs
namespace ChipmeoApis.Core.Entities;

public partial class Role
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public string? Description { get; set; }

    public string? DefaultRoute { get; set; }

    public bool? IsActive { get; set; }

    public DateTime? CreatedAt { get; set; }

    public virtual ICollection<Employee> Employees { get; set; } = new List<Employee>();

    public virtual ICollection<RolePermission> RolePermissions { get; set; } = new List<RolePermission>();
}




